### PR TITLE
PCHR-2140: Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by AkA84**
First draft of guidelines for contributors to the repo

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Contributing to the Shoreditch theme
+First of all, thank you for considering contributing to the Shoreditch theme!
+
+## Things to know before you start
+1. Bear in mind that the theme is currently in an **alpha stage** and at such many (even fundamental) things can be subject to change!
+2. Please read the [README](https://github.com/compucorp/org.civicrm.shoreditch/blob/staging/README.md) to get an overview of what the theme is and how it works
```
Comment (meladawy) [inline CONTRIBUTING.md]: I found some lines ends with "."  and some others doesn't. please try to make it consistent

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+TBD
+
+## Guidelines for `bootstrap.css`
+While `custom-civicrm.css` can be thought of as the "legacy mode" of Shoreditch, this component can be considered as the actual theme.
+
+There are three main types of changes you might want to introduce to the theme
+1. **Tweak the overall look-and-feel.** *(i.e., color palette, border radius, line height)* These are usually accomplished by working on the `overrides/_variables.scss` partial, as Bootstrap "exposes" many basic traits of the style as SASS variables.
+2. **Customize a Bootstrap component** *(i.e., changing the padding of `.panel`, adding a custom variation of `.btn`)* You  can do that in the `overrides/_variables.scss` file, or if Bootstrap doesn't provide a SASS variable for the change you need to implement, you can write the overriding style in the `overrides/style/_{component}.scss` partial.
+3. **Add a custom component** *(i.e., wizard component)* There are some component in CiviCRM that do not have any equivalent in the Bootstrap component library, and as such they have to be written from scratch. Those components go in the `components/` folder, each in an individual partial
```
Comment (meladawy) [inline CONTRIBUTING.md]: This also doesn't has "." at the end. just make it consistent with others

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+For custom components, use the [BEM](http://getbem.com/) convention, making sure to prefix the block name with `crm_`. So `.crm_custom-component`, `.crm_foo__bar--baz`, etc.
+
+#### Add a custom component as a last resort
+Adding a custom components to achieve a specific design should be considered as a last resort, when no suitable Bootstrap component had been found that could either be used as-is or being augmented with a custom modifier.
+
+#### No ad-hoc changes
+The theme is meant to be as generic as possible, and as such no site/extension specific changes and/or components should be added to it.
+
+## Testing
+In order to make sure that your work does not introduce regression issues or unexpected changes, there are two tools that you are encouraged to use: [org.civicrm.styleguide](https://github.com/civicrm/org.civicrm.styleguide/), and [BackstopJS](https://github.com/garris/BackstopJS)
```
Comment (meladawy) [inline CONTRIBUTING.md]: Do we need the "," before "and"? also ending the line with "." is preferred

**Review by meladawy — CHANGES_REQUESTED** [review]

**Review by meladawy — APPROVED** [review]

**Comment by totten** [comment]
This is a good idea. :) I like the substance/topics being conveyed.

As a personal editorial opinion... I think of `CONTRIBUTING.md` as talking about Git/Github/JIRA workflow considerations. It might make sense to:
 * Split this into a couple files (`CODING.md` to discuss code-structure/design-guidelines/things-to-keep-in-mind; `TESTING.md` for the testing process; `CONTRIBUTING.md` for any Git/Github/JIRA workflow.)
 * Put hyperlinks in `README.md` and `CONTRIBUTING.md` to reference the other files.

The split is appealing for a few reasons:
 * When reading a new library, I often start from the README and want to understand the code-structure/design-guidelines first -- the idea of making a contribution hasn't crossed my mind yet.
 * The CONTRIBUTING.md is displayed in Github when you open a PR. At that point, they've already written a change Obviously, it's good to remind them about the general guidelines (because some people shoot from the hip), but their patience is fading, and a long doc will make their eyes glaze over.

However, those are just opinions -- if you look around at different FOSS projects, there's a variety of `CONTRIBUTING.md` styles, so I think you have a lot of discretion about writing it.

**Comment by AkA84** [comment]
hey @totten , thanks a lot for your feedback!

What I ended up doing is something in the middle, I split the content in three different files as you suggested, but left CONTRIBUTING.md as the "entry point" from the README for people who want to contribute to the project (as that seems the common pattern)

I also added a section about how to report bugs and suggest enhancement, and some simple Git commit message guidelines

Let me know what you think, thanks!

**Review by meladawy — APPROVED** [review]
I feel more comfortable about this splitting. I approve this.

**Comment by totten** [comment]
Yeah, the split looks better, and I like how it moved the `npm`/`gulp` instructions to `CODING.md`.

**Pull request merged**